ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/network/DefaultP2PNetwork.java

```diff
@@ -325,29 +321,37 @@ public boolean removeMaintainedConnectionPeer(final Peer peer) {
   }
 
   @VisibleForTesting
-  Optional<DNSDaemon> getDnsDaemon() {
-    return Optional.ofNullable(dnsDaemon);
+  DNSPeerDiscoveryService getDnsDiscoveryService() {
+    return dnsDiscoveryService;
   }
 
-  @VisibleForTesting
-  DNSDaemonListener createDaemonListener() {
-    return (seq, records) -> {
-      final List<DiscoveryPeer> peers = new ArrayList<>();
-      for (final EthereumNodeRecord enr : records) {
-        final EnodeURL enodeURL =
-            EnodeURLImpl.builder()
-                .ipAddress(enr.ip())
-                .nodeId(enr.publicKey().bytes())
-                .discoveryPort(Optional.ofNullable(enr.udp()))
-                .listeningPort(Optional.ofNullable(enr.tcp()))
-                .build();
-        final DiscoveryPeer peer = DiscoveryPeer.fromEnode(enodeURL);
-        peers.add(peer);
-        rlpxAgent.connect(peer);
+  Consumer<NodeRecord> createNodeRecordListener() {
+    return enr -> {
+      final Optional<InetSocketAddress> tcp = enr.getTcpAddress();
+      final Optional<InetSocketAddress> udp = enr.getUdpAddress();
+      if (tcp.isEmpty() && udp.isEmpty()) {
+        return;
       }
-      // only replace dnsPeers if the lookup was successful:
-      if (!peers.isEmpty()) {
-        dnsPeers.set(peers);
+      final InetSocketAddress host = tcp.orElseGet(udp::get);
+      final Bytes compressedKey = (Bytes) enr.get(EnrField.PKEY_SECP256K1);
+      // final SignatureAlgorithm signatureAlgorithm =
+      //     SignatureAlgorithmType.DEFAULT_SIGNATURE_ALGORITHM_TYPE.get();
+      // final ECPoint decompressedPoint =
+      //
+      // signatureAlgorithm.publicKeyAsEcPoint(signatureAlgorithm.createPublicKey(compressedKey));
+      final ECPoint decompressedPoint = SECP256K1_CURVE.decodePoint(compressedKey.toArrayUnsafe());
```

Comment:
you might want to test for the case where the key is missing or invalid, and handle expliclity issues arising in this lambda.